diff --git a/node_modules/tempfile/index.js b/node_modules/tempfile/index.js
index 5a44ea6..ba4fff8 100644
--- a/node_modules/tempfile/index.js
+++ b/node_modules/tempfile/index.js
@@ -1,6 +1,7 @@
 'use strict';
 const path = require('path');
 const uuid = require('uuid');
-const tempDir = require('temp-dir');
+//const tempDir = require('temp-dir');
+const tempDir = process.env.TMPDIR;
 
 module.exports = ext => path.join(tempDir, uuid.v4() + (ext || ''));
